### ISSUE TITLE
Add support for RFC 7009 (revoking)

### ISSUE THREAD

**Issue by bakura10** [simplabs/ember-simple-auth#228]
Hi,

There is a spec now for OAuth 2 token revocation: https://tools.ietf.org/html/rfc7009

I think Ember-Simple-Auth should allow to have an option to activate it. The idea would be as follow:
- When explicitly logout, if option is enabled, it should issues a revocation request to the authorization server of both the access and refresh token.
- When Ember-Simple-Auth refreshes an access token by using a refresh token, it should revoke the old access token once it has received the new one.

The spec is quite simple: you just need to do a POST request with those two parameters:
- token: token to revoke
- token_type_hint: must be either "access_token" or "refresh_token"

I'm implementing the spec as part of the next version of ZfrOAuth2Server (PHP OAuth lib)

**Comment by Turbo87** [comment]
@bakura10 @marcoow maybe I'm reading it all wrong but https://tools.ietf.org/html/rfc7009#section-2.1 states that the **Client** credentials should be put in the `Authorization` header. It seems that the current implementation is using `Bearer <access_token>`, which doesn't seems to be covered by the spec. Is that a bug? Am I missing something here?

**Comment by marcoow** [comment]
The revoked token is sent in the request body actually. Client credentials generally make no sense for public clients because as soon as you deploy the client to the internet the credentials are public and cannot be trusted anymore.

**Comment by Turbo87** [comment]
@marcoow I was not talking about the token that should be revoked, I meant the token that is used to authenticate with the API server. I understand that Client credentials don't make any sense for public clients, but the spec still seems to specify that behavior.

tl;dr

current request: `Authorization: Bearer <access_token>` + Body: `token=<refresh_token>`
spec request: `Authorization: Basic <client_credentials>` + Body: `token=<refresh_token>` without any way of specifying the `access_token`...

**Comment by marcoow** [comment]
Yeah, that's true - the bearer token shouldn't be sent with that request.

**Comment by Turbo87** [comment]
on second thought I'm not even sure anymore if that was currently the case... my head is spinning from all that OAuth crazyness ;)

**Comment by Turbo87** [comment]
thanks for the quick reply though :+1:
